src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>ordered_before</th>
+      <th>global_popularity</th>
+      <th>abandoned_before</th>
+      <th>normalised_price</th>
+      <th>product_type</th>
+      <th>vendor</th>
+      <th>created_at_year</th>
+      <th>created_at_month</th>
+      <th>created_at_day</th>
+      <th>created_at_hour</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>0</td>
+      <td>0.000000</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>ricepastapulses</td>
+      <td>clearspring</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>5</td>
+      <td>16</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>0</td>
+      <td>0.000000</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>ricepastapulses</td>
+      <td>clearspring</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>5</td>
+      <td>17</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>0</td>
+      <td>0.000000</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>ricepastapulses</td>
+      <td>clearspring</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>5</td>
+      <td>20</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>ricepastapulses</td>
+      <td>clearspring</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>8</td>
+    </tr>
+    <tr>
+      <th>5</th>
+      <td>0</td>
+      <td>0.038462</td>
+      <td>0</td>
+      <td>0.081052</td>
+      <td>ricepastapulses</td>
+      <td>clearspring</td>
+      <td>2020</td>
+      <td>10</td>
+      <td>6</td>
+      <td>10</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Now, for the categorical variables (product_type and vendor), which are strings, my first idea was to use Dummy variables for all, which are similar to one-hot encoding but avoids one redundant category by dropping the first category.
+
+But first, I check how many categories each column contains to address if it is feasable to do dummy variables or if too many columns would be created.
+
+
+```python
+for col in categorical_cols:
+    print("Number of cateogories in: ", col, ": ", len(X_train_cat[col].unique()))  # Returns array of unique values
+
+```
+
+    Number of cateogories in:  product_type :  59
+    Number of cateogories in:  vendor :  240
+
+
+We can observe that the only column where using the Dummy variables approach may be feasible is for product_type. Furthermore, since  vendor would create too many columns with this approach (as it has many different variables), I try One Hot encoder (with colisions).
```

Comment:
You are not using one hot encoding. You are just replacing the stirngs by a number. I would argue against this approach:

 - Is the ordering important in the linear regrssion? If so, how are we ordering the categories?
 - Are teh distance between the categories relevant? If so, how are we dealing with it?